packages/relay-compiler/graphql-compiler/transforms/DefinitionRenameTransform.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const shortHash = require('short-hash');
+const resolve = require('resolve');
+
+function transform(doc, filePath) {
+  return {
+    ...doc,
+    definitions: doc.definitions.map(def => ({
+      ...def,
+      metadata: {
```

Comment:
My first attempt was to do this with the rest of the `IRTransforms`, but there are assumptions of unique fragment names before those transforms are executed. This strategy is not ideal either because it is adding a new `metadata` property to objects that are typed externally by `graphql-js`. Do you think it would be better for this to happen in [`ASTConvert`](https://github.com/facebook/relay/blob/master/packages/relay-compiler/graphql-compiler/core/ASTConvert.js) or somewhere else?